### PR TITLE
build: produce a .zip for macOS distribution

### PR DESCRIPTION
**Pull request by fanquake**
Reviving the discussion around using a `.zip` for the distributed macOS binaries, as opposed to a `.dmg`.

Given we only had a single report of the "no finder window" issue (#26176), I wonder if that means macOS users were able to figure it out, they gave up/didn't report, or, we just have very few macOS users.

Related to #18128.

### CODE REVIEW

**Comment by prusnak** [comment]
Concept ACK

Here's a revised version of the passage with improved grammar and natural-sounding language:

At present, the user experience with DMG is terrible (as noted in this issue: https://github.com/bitcoin/bitcoin/issues/26176). If there's no straightforward solution on the horizon, it would be best to switch to using ZIP instead. Using DMG without a Finder window provides no advantages and only leads to a lot of confusion. As evidence, Visual Studio Code has recently started distributing their macOS builds as ZIP files instead of DMGs, as you can see on their download page: https://code.visualstudio.com/Download.

**Comment by RandyMcMillan** [comment]
Concept ACK

tested on macOS 12.6.1 (21G217)

**Comment by Sjors** [comment]
Concept ACK, I second @prusnak.

**Comment by dergoegge** [comment]
Concept ACK

**Comment by fanquake** [comment]
Rebased past #27172.

**Comment by fanquake** [comment]
A few Concept ACKs here, so have rebased and undrafted.

**Comment by Sjors** [comment]
I'd like to test the Guix build too. @achow101 can you sign it? Or is there a way to self-sign it?

File: contrib/guix/manifest.scm [hunk 1]
```diff
@@ -5,7 +5,6 @@
              (gnu packages bash)
              (gnu packages bison)
              (gnu packages certs)
-             (gnu packages cdrom)
```
Comment (Sjors) [inline contrib/guix/manifest.scm:8]: 💿

**Review by Sjors — COMMENTED** [review]
Code review ACK c5342f0877b663e028c0acd807daf4d1e02b46e5

**Comment by RandyMcMillan** [comment]
note:
a bundled README should be in README.txt format - we can't assume the user can open .md files by default.

**Comment by fanquake** [comment]
> we can't assume the user can open .md files by default.

I'm pretty sure we can assume that on macOS.

**Comment by fanquake** [comment]
Rebased, and added a number of small doc fixups.

**Comment by fanquake** [comment]
> I end up with three files / folders:

This is the same as the current build process, where you end up with:

1. Bitcoin-Qt.app (which Finder displays as "Bitcoin Core")
2. Bitcoin-Core.dmg
3. dist/Bitcoin-Qt.app

Happy to refine/rework this further in a follow up, but don't really want to start changing this now, as it's not new behaviour.

**Comment by hebasto** [comment]
> The commits can be improved, but currently good enough for (non-guix) testing / discussion.

What are plans regarding Guix builds?

**Comment by hebasto** [comment]
> > What are plans regarding Guix builds?
> 
> What you've quoted was just an outdated part of the PR description, that I've now dropped.

Well, then Guix does not produced unsigned `*-apple-darwin-unsigned.zip`, instead of `*-apple-darwin-unsigned.dmg`:
```
1bed850e74f985e3db277160529aff767f4d3be00822ee8e204cdff304664727  guix-build-bec052f9ac0b/output/arm64-apple-darwin/SHA256SUMS.part
6b3a93b33d163dcc6a1a8582c3db1fe2c828620757c078e0c59f3e1b1eee3990  guix-build-bec052f9ac0b/output/arm64-apple-darwin/bitcoin-bec052f9ac0b-arm64-apple-darwin-unsigned.tar.gz
6c0b10abc82fa15aa54f16bafb4d32680ac37e1f4fd8ca9a73ecd2b1c57198c9  guix-build-bec052f9ac0b/output/arm64-apple-darwin/bitcoin-bec052f9ac0b-arm64-apple-darwin.tar.gz
a3b761f21252bdb53e397b92658f95512f8bf030a464b8ec153dd87796bcb328  guix-build-bec052f9ac0b/output/dist-archive/bitcoin-bec052f9ac0b.tar.gz
```

**Comment by hebasto** [comment]
> > Guix does not produce unsigned
> 
> Fixed.

Now, it needs to be reproducible:

- build `#1`:
```
8f0fbd31364e2bc589be140372c56a71d8db0ccb06118923a8a6f461c65728ab  guix-build-414e840c297f/output/arm64-apple-darwin/SHA256SUMS.part
8aa7c8fed1f8cc7f2d4735b87c4f57c6383191d71b10553729604863beda7e01  guix-build-414e840c297f/output/arm64-apple-darwin/bitcoin-414e840c297f-arm64-apple-darwin-unsigned.tar.gz
ea3f6623b130cf8da7d877e8828d607ca1a72654811da8fd3a840388e27cfe19  guix-build-414e840c297f/output/arm64-apple-darwin/bitcoin-414e840c297f-arm64-apple-darwin-unsigned.zip
14812ef0488cbb6e4d442f086472dcc5f2f7fac0c469d5d0c9f0e2d99b0b8c55  guix-build-414e840c297f/output/arm64-apple-darwin/bitcoin-414e840c297f-arm64-apple-darwin.tar.gz
3dca7457cb512cd812b62d21bc7849f935e8444cd8e05982d039eff174f9ef31  guix-build-414e840c297f/output/dist-archive/bitcoin-414e840c297f.tar.g
```
- build `#2`:
```
308e2ca63883a7c710543701ddad60b5a6640882800604a8c4fa8827a7e810d1  guix-build-414e840c297f/output/arm64-apple-darwin/SHA256SUMS.part
8aa7c8fed1f8cc7f2d4735b87c4f57c6383191d71b10553729604863beda7e01  guix-build-414e840c297f/output/arm64-apple-darwin/bitcoin-414e840c297f-arm64-apple-darwin-unsigned.tar.gz
d4c166747d7478afc3859650eea289fa73b66a8a6f7bee28501912a628b2149e  guix-build-414e840c297f/output/arm64-apple-darwin/bitcoin-414e840c297f-arm64-apple-darwin-unsigned.zip
14812ef0488cbb6e4d442f086472dcc5f2f7fac0c469d5d0c9f0e2d99b0b8c55  guix-build-414e840c297f/output/arm64-apple-darwin/bitcoin-414e840c297f-arm64-apple-darwin.tar.gz
3dca7457cb512cd812b62d21bc7849f935e8444cd8e05982d039eff174f9ef31  guix-build-414e840c297f/output/dist-archive/bitcoin-414e840c297f.tar.gz
```

File: Makefile.am [hunk 2]
```diff
@@ -124,23 +124,23 @@ osx_volname:
...
 
-deploydir: $(OSX_DMG)
+deploydir: $(OSX_ZIP)
 else !BUILD_DARWIN
 APP_DIST_DIR=$(top_builddir)/dist
 
-$(OSX_DMG): deploydir
-	$(XORRISOFS) -D -l -V "$(OSX_VOLNAME)" -no-pad -r -dir-mode 0755 -o $@ $(APP_DIST_DIR) -- $(if $(SOURCE_DATE_EPOCH),-volume_date all_file_dates =$(SOURCE_DATE_EPOCH))
+$(OSX_ZIP): deploydir
+		zip -r $@ $(APP_DIST_DIR)
```
Comment (hebasto) [inline Makefile.am:135]: 8d8d4325fedad5c140ea1ad4568f19609c431e2e

The following diff is intended to fix the [reproducibility](reproducible) issue:
```suggestion
	if [ -n "$(SOURCE_DATE_EPOCH)" ]; then find $(APP_DIST_DIR) -exec touch -d @$(SOURCE_DATE_EPOCH) {} +; fi
	zip -X -r $@ $(APP_DIST_DIR)
```

**Comment by hebasto** [comment]
Guix build on `arm64`:
```
3bdff2fec75d06d588b91ae89c94cbeea474d11e4c03d0b2bbf89de8565b8f92  guix-build-18df951ef677/output/arm64-apple-darwin/SHA256SUMS.part
c0c0e06b70e703eb47f2e9b6ae90b036af8585acba665a2ebd3aa9bbeb0159f0  guix-build-18df951ef677/output/arm64-apple-darwin/bitcoin-18df951ef677-arm64-apple-darwin-unsigned.tar.gz
3f0772f177a655e5d8b41bdd9355dec47100de9490d12335d13661197d571b7d  guix-build-18df951ef677/output/arm64-apple-darwin/bitcoin-18df951ef677-arm64-apple-darwin-unsigned.zip
d775d4564758b92f7517c240b3bb65d228ab9f19f8d848d8396f6624ecd41fed  guix-build-18df951ef677/output/arm64-apple-darwin/bitcoin-18df951ef677-arm64-apple-darwin.tar.gz
93c6d27d0a4264c5f66ceeafdd49967dbfab28561db9df5f3e9b8e7de912e970  guix-build-18df951ef677/output/dist-archive/bitcoin-18df951ef677.tar.gz
ba3c3aa83ffb72adb20e38191bd2aa5e9f0ec4f378f6adbabf08bcc601b6c2f7  guix-build-18df951ef677/output/x86_64-apple-darwin/SHA256SUMS.part
cde75ea3fd7c8d5227d41723909e2078cfe747ee829269755863ef629a819b95  guix-build-18df951ef677/output/x86_64-apple-darwin/bitcoin-18df951ef677-x86_64-apple-darwin-unsigned.tar.gz
caa6d10eada1f282447a7e885cda3132b4e73fc9cf37ba01eb6efd0d7060cf45  guix-build-18df951ef677/output/x86_64-apple-darwin/bitcoin-18df951ef677-x86_64-apple-darwin-unsigned.zip
b21585791849cc6ef741f318fe041c1c4bd1eaa84e3e255f75989be66f05876a  guix-build-18df951ef677/output/x86_64-apple-darwin/bitcoin-18df951ef677-x86_64-apple-darwin.tar.gz
```

**Comment by fanquake** [comment]
```bash
cbe2109a1e88b431f462f6b8a62e785fe1f5f6249da55f984129c2a2ef454503  guix-build-23d89518e907/output/arm64-apple-darwin/SHA256SUMS.part
47c72d2c1c4ec35cd72db66cd602a213b18d43274dc3ab3fa8065f05e8844f83  guix-build-23d89518e907/output/arm64-apple-darwin/bitcoin-23d89518e907-arm64-apple-darwin-unsigned.tar.gz
eeeccfef61a67f39741e153b392cadc3e44ac317fd0608b30fcd8aab206cd2d2  guix-build-23d89518e907/output/arm64-apple-darwin/bitcoin-23d89518e907-arm64-apple-darwin-unsigned.zip
a662ab05fd44ef5c08a89d8d140d0bb66c8e53642c11222aef49cdd5e3a86321  guix-build-23d89518e907/output/arm64-apple-darwin/bitcoin-23d89518e907-arm64-apple-darwin.tar.gz
bc4b3bfdd24b64fb3decb582ee4b32bfa4b15ffdd23753a73135f7a5977cbf4a  guix-build-23d89518e907/output/dist-archive/bitcoin-23d89518e907.tar.gz
2338ff8dc38286b169b8efc213eae7e62481d756e01056245f55126dc97dbc8e  guix-build-23d89518e907/output/x86_64-apple-darwin/SHA256SUMS.part
c6cd0dbcb3eb8dc79c2e78b0e954271dfa724d9523c3338c8475a46bc1d2efd4  guix-build-23d89518e907/output/x86_64-apple-darwin/bitcoin-23d89518e907-x86_64-apple-darwin-unsigned.tar.gz
5a31c1b115c5e2d384d330e25633ea661d5c2a20acafe0cad95e89faafad84e4  guix-build-23d89518e907/output/x86_64-apple-darwin/bitcoin-23d89518e907-x86_64-apple-darwin-unsigned.zip
fa7954064abdedb2ee0e6c83cad3f8a86a1a87c447f7a3515ccd8da2ec244334  guix-build-23d89518e907/output/x86_64-apple-darwin/bitcoin-23d89518e907-x86_64-apple-darwin.tar.gz
```

**Comment by hebasto** [comment]
Guix build on `arm64`:
```
a9eb04f494db558318b52ec609d67478c761b281902204cc1ebfb49450216ac9  guix-build-23d89518e907/output/arm64-apple-darwin/SHA256SUMS.part
47c72d2c1c4ec35cd72db66cd602a213b18d43274dc3ab3fa8065f05e8844f83  guix-build-23d89518e907/output/arm64-apple-darwin/bitcoin-23d89518e907-arm64-apple-darwin-unsigned.tar.gz
04651cc05a9ae9786ecdba7c44f6823559e70aeb75d84875907c61e0ad137cdc  guix-build-23d89518e907/output/arm64-apple-darwin/bitcoin-23d89518e907-arm64-apple-darwin-unsigned.zip
a662ab05fd44ef5c08a89d8d140d0bb66c8e53642c11222aef49cdd5e3a86321  guix-build-23d89518e907/output/arm64-apple-darwin/bitcoin-23d89518e907-arm64-apple-darwin.tar.gz
bc4b3bfdd24b64fb3decb582ee4b32bfa4b15ffdd23753a73135f7a5977cbf4a  guix-build-23d89518e907/output/dist-archive/bitcoin-23d89518e907.tar.gz
1f91c2686d2973013cba97e67b4b9f68430b5db05d21869fee19447d21cfc85d  guix-build-23d89518e907/output/x86_64-apple-darwin/SHA256SUMS.part
c6cd0dbcb3eb8dc79c2e78b0e954271dfa724d9523c3338c8475a46bc1d2efd4  guix-build-23d89518e907/output/x86_64-apple-darwin/bitcoin-23d89518e907-x86_64-apple-darwin-unsigned.tar.gz
807cd39d020c344618bd0e7567c9de55a44a8280ddde78a7879c7fa1cc2253a2  guix-build-23d89518e907/output/x86_64-apple-darwin/bitcoin-23d89518e907-x86_64-apple-darwin-unsigned.zip
fa7954064abdedb2ee0e6c83cad3f8a86a1a87c447f7a3515ccd8da2ec244334  guix-build-23d89518e907/output/x86_64-apple-darwin/bitcoin-23d89518e907-x86_64-apple-darwin.tar.gz
```

File: Makefile.am [hunk 3]
```diff
@@ -124,23 +124,24 @@ osx_volname:
...
-deploydir: $(OSX_DMG)
+deploydir: $(OSX_ZIP)
 else !BUILD_DARWIN
 APP_DIST_DIR=$(top_builddir)/dist
 
-$(OSX_DMG): deploydir
-	$(XORRISOFS) -D -l -V "$(OSX_VOLNAME)" -no-pad -r -dir-mode 0755 -o $@ $(APP_DIST_DIR) -- $(if $(SOURCE_DATE_EPOCH),-volume_date all_file_dates =$(SOURCE_DATE_EPOCH))
+$(OSX_ZIP): deploydir
+	if [ -n "$(SOURCE_DATE_EPOCH)" ]; then find $(APP_DIST_DIR) -exec touch -d @$(SOURCE_DATE_EPOCH) {} +; fi
+	zip -X -r $@ $(APP_DIST_DIR)
```
Comment (hebasto) [inline Makefile.am:136]: Unfortunately, using the `-r` option still introduces non-reproducibilty. Diffoscope points at some metadata related to directories. Not sure, but it looks like they are about creation timestamps.

Suggesting to use the same approach as we are using for Windows builds:
```suggestion
	find $(APP_DIST_DIR) | sort | zip -X@ $@
```

Two subsequent runs on my Ubuntu 22.04 returned the same hashes.

**Comment by hebasto** [comment]
> As evidence, Visual Studio Code has recently started distributing their macOS builds as ZIP files instead of DMGs, as you can see on their download page: [code.visualstudio.com/Download](https://code.visualstudio.com/Download).

In comparison to UX of downloading `VSCode-darwin.zip`, in the current PR branch, there is a top directory named `dist`, which seems redundant:

image

**Comment by hebasto** [comment]
Someone on Twitter drew attention to potential security implications, for instance:
- [New ZIP domains spark debate among cybersecurity experts](https://www.bleepingcomputer.com/news/security/new-zip-domains-spark-debate-among-cybersecurity-experts/)
- [Don't Click That ZIP File! Phishers Weaponizing .ZIP Domains to Trick Victims](https://thehackernews.com/2023/05/dont-click-that-zip-file-phishers.html)

File: Makefile.am (hunk 3 above)
Comment (fanquake) [reply Makefile.am]: Applied.

**Comment by hebasto** [comment]
Guix build on `x86_64`:
```
7851f08737a0133002c0224d4bd5e52aed7eaee91adac46f1247ef164775d037  guix-build-23d39c5079cf/output/arm64-apple-darwin/SHA256SUMS.part
172e378d18cd9c624e4cdce51228bfbb43d25098871c18911c70eeb1c2390f34  guix-build-23d39c5079cf/output/arm64-apple-darwin/bitcoin-23d39c5079cf-arm64-apple-darwin-unsigned.tar.gz
3de4d2cec142a8464788da7571b2f09a47893c7f8d75fe52912ecfcefac72242  guix-build-23d39c5079cf/output/arm64-apple-darwin/bitcoin-23d39c5079cf-arm64-apple-darwin-unsigned.zip
33da433dbd3c3d642756a62d19da37270a18a13dd2dfe9cb40327a510f722b7f  guix-build-23d39c5079cf/output/arm64-apple-darwin/bitcoin-23d39c5079cf-arm64-apple-darwin.tar.gz
1f30f314a56781944dd3fd05eef8604afe9cf300361381f157b197e349501da8  guix-build-23d39c5079cf/output/dist-archive/bitcoin-23d39c5079cf.tar.gz
f14dae2b0fb901a2fc4d70d8af58bc83832a3f2d7dd238db6e0e4a7e77888c09  guix-build-23d39c5079cf/output/x86_64-apple-darwin/SHA256SUMS.part
085b59e69556d2f89cc6f662fc4d8da94b01d3c7d47b414128c0f5f6ee38b8fc  guix-build-23d39c5079cf/output/x86_64-apple-darwin/bitcoin-23d39c5079cf-x86_64-apple-darwin-unsigned.tar.gz
f7f1161609ca379ec60002d684127e831ed9e72404ce4202262f2bb01bc77d96  guix-build-23d39c5079cf/output/x86_64-apple-darwin/bitcoin-23d39c5079cf-x86_64-apple-darwin-unsigned.zip
fb19c22a2e31de8189b2696a8d457902423c40878dc4c031e75e9a9797d04c9a  guix-build-23d39c5079cf/output/x86_64-apple-darwin/bitcoin-23d39c5079cf-x86_64-apple-darwin.tar.gz
```

Matched to @fanquake's [one](https://github.com/bitcoin/bitcoin/pull/27099#issuecomment-1611235566).

**Comment by prusnak** [comment]
`zip foo.zip dist` should read `zip foo.zip dist/*` to drop the top-level `dist` folder

**Comment by hebasto** [comment]
Tested `bitcoin-8a6275ba6ea6-x86_64-apple-darwin-unsigned.zip`.

I think a couple of comments still need to be addressed:
- https://github.com/bitcoin/bitcoin/pull/27099#issuecomment-1610057916
- https://github.com/bitcoin/bitcoin/pull/27099#issuecomment-1612862819

**Comment by Sjors** [comment]
tACK 8a6275ba6ea6e8eb5342442221d27d71670e0be8

Tested that it produces a working zip / binary for me on Intel macOS 13.5.1, both when using `make deploy`  on the machine itself, and when using Guix on Ubuntu (AMD).

```
fec40bc404855508466d4fdce4f729770b73bf882d9622dbd3505ae2673bb8f1  guix-build-8a6275ba6ea6/output/arm64-apple-darwin/SHA256SUMS.part
2edaca85d4a5f45f0c6b3c461dc90a878a5c672025a965c2f0452dedd722e1ed  guix-build-8a6275ba6ea6/output/arm64-apple-darwin/bitcoin-8a6275ba6ea6-arm64-apple-darwin-unsigned.tar.gz
d603238e2d5410125058a3810af296fc3d575bf86011feddc2c7ff41336098a7  guix-build-8a6275ba6ea6/output/arm64-apple-darwin/bitcoin-8a6275ba6ea6-arm64-apple-darwin-unsigned.zip
e882484df88c76dec6a0eb000735c7941831a4df98f86a4309839221618f6650  guix-build-8a6275ba6ea6/output/arm64-apple-darwin/bitcoin-8a6275ba6ea6-arm64-apple-darwin.tar.gz
367a361ccd3fa9bc6b8ac99eb23b42a6d4baed6746067f94f6fbb41b1cb5351d  guix-build-8a6275ba6ea6/output/dist-archive/bitcoin-8a6275ba6ea6.tar.gz
25353a797dd0e37d229414ff814692c7a2b6930b16e33d41b8e999f1777c92ab  guix-build-8a6275ba6ea6/output/x86_64-apple-darwin/SHA256SUMS.part
8cf69e85249762e04e9fdb707d9e50d37f9d9f6c7aa239477f5a5c913063201e  guix-build-8a6275ba6ea6/output/x86_64-apple-darwin/bitcoin-8a6275ba6ea6-x86_64-apple-darwin-unsigned.tar.gz
56c042d999726e8179f69b5d354bfcb82fcc69b1926f312e30c396d6752cffa1  guix-build-8a6275ba6ea6/output/x86_64-apple-darwin/bitcoin-8a6275ba6ea6-x86_64-apple-darwin-unsigned.zip
c99a9e1e01db20140855fb2dfe662d750e773bcd3aee2ee94f27905f4bd31b9d  guix-build-8a6275ba6ea6/output/x86_64-apple-darwin/bitcoin-8a6275ba6ea6-x86_64-apple-darwin.tar.gz
```

**Comment by hebasto** [comment]
Having an error when building Guix:
```
cd ./dist && find . | sort | zip -X@ ..//distsrc-base/distsrc-11f8b6b2b58e-x86_64-apple-darwin/output/bitcoin-11f8b6b2b58e-x86_64-apple-darwin-unsigned.zip
zip I/O error: No such file or directory
zip error: Could not create output file (..//distsrc-base/distsrc-11f8b6b2b58e-x86_64-apple-darwin/output/bitcoin-11f8b6b2b58e-x86_64-apple-darwin-unsigned.zip)
```

**Comment by fanquake** [comment]
Someone else can take this over if interested.

**Comment by Sjors** [comment]
Difference between my ACK and the last push:

```
$ git range-diff master 8a6275ba6ea6e8eb5342442221d27d71670e0be8 HEAD
1:  667d5735ce = 1:  bc95abd588 build: add -zip option to macdeployqtplus
2:  85f8152b47 ! 2:  b7e409b7bf build: produce a .zip for macOS distribution
    @@ Makefile.am: osx_volname:
     -  $(XORRISOFS) -D -l -V "$(OSX_VOLNAME)" -no-pad -r -dir-mode 0755 -o $@ $(APP_DIST_DIR) -- $(if $(SOURCE_DATE_EPOCH),-volume_date all_file_dates =$(SOURCE_DATE_EPOCH))
     +$(OSX_ZIP): deploydir
     +  if [ -n "$(SOURCE_DATE_EPOCH)" ]; then find $(APP_DIST_DIR) -exec touch -d @$(SOURCE_DATE_EPOCH) {} +; fi
    -+  find $(APP_DIST_DIR) | sort | zip -X@ $@
    ++  cd $(APP_DIST_DIR) && find . | sort | zip -X@ ../$@

      $(APP_DIST_DIR)/$(OSX_APP)/Contents/MacOS/Bitcoin-Qt: $(OSX_APP_BUILT) $(OSX_PACKAGING)
        INSTALL_NAME_TOOL=$(INSTALL_NAME_TOOL) OTOOL=$(OTOOL) STRIP=$(STRIP) $(PYTHON) $(OSX_DEPLOY_SCRIPT) $(OSX_APP) $(OSX_VOLNAME) -translations-dir=$(QT_TRANSLATION_DIR)
3:  117d01be1d = 3:  e78916bf9d macdeploy: remove DMG generation from deploy script
4:  8a6275ba6e = 4:  11f8b6b2b5 build: remove dmg dependencies
```

Getting the same (?) error as @hebasto now :-(

```
if [ -n "1668774980" ]; then find ./dist -exec touch -d @1668774980 {} +; fi
cd ./dist && find . | sort | zip -X@ ..//distsrc-base/distsrc-11f8b6b2b58e-x86_64-apple-darwin/output/bitcoin-11f8b6b2b58e-x86_64-apple-darwin-unsigned.zip
zip I/O error: No such file or directory
zip error: Could not create output file (..//distsrc-base/distsrc-11f8b6b2b58e-x86_64-apple-darwin/output/bitcoin-11f8b6b2b58e-x86_64-apple-darwin-unsigned.zip)
make: *** [Makefile:1304: /distsrc-base/distsrc-11f8b6b2b58e-x86_64-apple-darwin/output/bitcoin-11f8b6b2b58e-x86_64-apple-darwin-unsigned.zip] Error 15
```

If anyone takes this over, please ping me.

**Comment by hebasto** [comment]
Picked in https://github.com/bitcoin/bitcoin/pull/28432.

**Pull request closed without merging**